### ISSUE TITLE
Occasionally crashes on onSaveInstanceState()

### ISSUE THREAD

**Issue by ypresto** [ypresto/RecyclerView-AbsoluteLayoutManager#7]
Currently onSaveInstanceState() returns empty state if `getItemCount() == 0`.
But we face crashes caused by out-of-sync state of LayoutProvider: `LayoutProvider.mState.mItemCount == 0` while `getItemCount() > 0`.

```
Fatal Exception: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
       at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
       at java.util.ArrayList.get(ArrayList.java:308)
       at us.mitene.app.album.ui.AlbumThumbnailLayoutProvider.getLayoutAttributeForItemAtPosition(Unknown Source:144)
       at net.ypresto.recyclerview.absolutelayoutmanager.AbsoluteLayoutManager.net.ypresto.recyclerview.absolutelayoutmanager.AnchorHelper.calculateAnchorItemInRect(Unknown Source:17041)
       at us.mitene.app.album.ui.MonthlyAlbumFragment.onSaveInstanceState(Unknown Source:205)
```

onItemsChanged() or its family methods are lazy fired inside of dispatchLayout(). This can cause out-of-sync with following step:
1. Change adapter content and call notifyDataSetChanged() or etc.
2. Call prepareLayoutProvider() from onSaveInstanceState() **but no update is performed**.
3. onItemsChanged() is called from dispatchLayout() and `mIsLayoutProviderDirty` is set to true.
4. prepareLayoutProvider() is called from onLayoutChildren() and provider state is updated at last.

**Comment by ypresto** [comment]
Fix candidate: https://github.com/ypresto/RecyclerView-AbsoluteLayoutManager/tree/fix-crash-on-save

**Comment by ypresto** [comment]
Fixed in v0.2.1
